### PR TITLE
First google benchmarks for hierarchy components.

### PR DESCRIPTION
**Pull request by AMZN-Olex**
Already proving useful by pointing out a performance defect. (I have fixes for that that will go in a separate PR.)

```
Run on (32 X 3394 MHz CPU s)
CPU Caches:
  L1 Data 32K (x16)
  L1 Instruction 32K (x16)
  L2 Unified 524K (x16)
  L3 Unified 33554K (x2)
---------------------------------------------------------------------------------------------------------------
Benchmark                                                                     Time             CPU   Iterations
---------------------------------------------------------------------------------------------------------------
ServerDeepHierarchyBenchmark/RebuildHierarchy                              43.7 us         43.9 us        16000
ServerDeepHierarchyBenchmark/RebuildHierarchyRemoveAndAddLastChild          133 us          134 us         5600
ServerDeepHierarchyBenchmark/RebuildHierarchyRemoveAndAddMiddleChild        167 us          165 us         4073
ServerDeepHierarchyBenchmark/RebuildHierarchyRemoveAndAddFirstChild         231 us          225 us         2987
OKAY AzRunBenchmarks() returned 0
```

Signed-off-by: AMZN-Olex <5432499+AMZN-Olex@users.noreply.github.com>

### CODE REVIEW

**Review by AMZN-puvvadar — APPROVED** [review]
I'm really liking how testable this suite of code is!

File: Gems/Multiplayer/Code/Tests/ServerHierarchyBenchmarks.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        {
+            HierarchyBenchmarkBase::internalSetUp();
+
+            m_root = AZStd::make_unique<EntityInfo>((1), "root", RootNetEntityId, EntityInfo::Role::Root);
+            CreateParent(*m_root);
+
+            m_children = AZStd::make_unique<AZStd::vector<AZStd::shared_ptr<EntityInfo>>>();
+
+            EntityInfo* parent = m_root.get();
+            for (int i = 0; i < 15; ++i)
```
Comment (lmbr-pip) [inline Gems/Multiplayer/Code/Tests/ServerHierarchyBenchmarks.cpp:35]: nit: I would call out the requirement that 16 is a deep as hierarchy can go now, its not clear to future maintainers why 15 is the stopping condition here.

**Review by lmbr-pip — APPROVED** [review]
LG

File: Gems/Multiplayer/Code/Tests/ServerHierarchyBenchmarks.cpp (hunk 1 above)
Comment (AMZN-Olex) [reply Gems/Multiplayer/Code/Tests/ServerHierarchyBenchmarks.cpp:35]: Sounds good, will do.

**Pull request merged**